eng/common/tools.ps1

```diff
@@ -814,7 +814,8 @@ function MSBuild-Core() {
       Write-Host "See log: $buildLog" -ForegroundColor DarkGray
     }
 
-    if ($ci) {
+    # When running on Azure Pipelines, override the returned exit code to avoid double logging.
+    if ($ci -and $env:BUILD_BUILDNUMBER -ne $null) {
```

Comment:
I don't think the "-ne $null" is necessary.  Powershell is pretty good at truthy checks.  That will also help if, for some reason it's like, the empty string or something.

Reply:
Ah. If it's a consistent pattern, then you can keep it. I was just worried about empty strings triggering a false positive here can the wrong behavior coming up.